ci-operator/config/openshift/ocs-operator/openshift-ocs-operator-master.yaml

```diff
@@ -1,16 +1,56 @@
-build_root:
-  image_stream_tag:
-    cluster: https://api.ci.openshift.org
-    name: release
-    namespace: openshift
-    tag: golang-1.12
+base_images:
+  os:
+    name: ubi
+    namespace: ocp
+    tag: '8'
+  operator-registry:
+    name: "4.2"
+    namespace: ocp
+    tag: operator-registry
+
 resources:
   '*':
     limits:
       memory: 4Gi
     requests:
       cpu: 100m
       memory: 200Mi
+
+images:
+- from: os
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/openshift/ocs-operator/build/_output/bin/ocs-operator
+  dockerfile_path: openshift-ci/Dockerfile.deploy
+  to: ocs-operator
+- from: operator-registry
+  dockerfile_path: openshift-ci/Dockerfile.registry.intermediate
+  to: operator-registry-base
```

Comment:
I see the intermediary Dockerfile being used here. Why do we need to have a intermediary image for the registry image?

Reply:
this link explains it well. https://github.com/redhat-developer/devconsole-operator/tree/master/openshift-ci#operator-registry-base

The intermediate image is part of a trick used to be able to accurately predict the ocs-operator container image url that is being built by this same job. The intermediate build has the unique `$OPENSHIFT_BUILD_NAMESPACE` propagated into it that is related to this CI job. By basing the ocs-registry container off of the intermediate container, we are able to use that `OPENSHIFT_BUILD_NAMESPACE` var to predict the url of the ocs-operator image which is injected into the CSV. 

Basically, it's just a way to have the ocs-registry CSV reference the ocs-operator image correctly. otherwise the ocs-registry would point to the quay.io/ocs-dev/ocs-operator:latest image instead of the ocs-operator container built by this CI job.

Also, I didn't come up with this pattern. Several other projects are using this method to solve the exact same issue.

Reply:
Thanks for the explanation! I'd assumed this to be a pattern used elsewhere, but wanted to know why.